pkg/freeze/containerd/pause.go

```diff
@@ -89,27 +98,57 @@ func lookupContainerID(ctx context.Context, conn *grpc.ClientConn, podUID, conta
 		},
 	})
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	if len(pods.Items) == 0 {
-		return "", fmt.Errorf("pod %s not found", podUID)
+		return nil, fmt.Errorf("pod %s not found", podUID)
 	}
 	pod := pods.Items[0]
 
 	ctrs, err := client.ListContainers(ctx, &cri.ListContainersRequest{Filter: &cri.ContainerFilter{
 		PodSandboxId: pod.Id,
-		LabelSelector: map[string]string{
-			"io.kubernetes.container.name": containerName,
-		},
 	}})
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+
+	return ctrs, nil
+}
+
+// Pause performs a pause action on a specific container
+func (c *ContainerdCri) Pause(ctx context.Context, ctrd *containerd.Client, containerList []string) error {
```

Comment:
I think I'd have this interface take a single container, it'll make it easier to eventually change/test how we treat failures of individual containers etc, and it's nice if our wrapper of containerd is minimal